storage: TestLeaseNotUsedAfterRestart failed under stress [skipped]

SHA: https://github.com/cockroachdb/cockroach/commits/6885730c58f9a45511f92be95e94129005d6b875

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1100026&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190118 08:07:26.534361 72683 storage/store.go:1465  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190118 08:07:26.551490 72597 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:33767" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190118 08:07:26.610034 72597 storage/client_replica_test.go:1126  restarting
I190118 08:07:26.619282 72597 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
W190118 08:07:26.621769 72370 internal/client/txn.go:520  [s1,r5/1:/{System/tse-Table/System…}] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190118 08:07:26.622059 72370 storage/node_liveness.go:773  [s1,r5/1:/{System/tse-Table/System…}] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
I190118 08:07:26.624330 72597 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] storage.replicate: processing replica
I190118 08:07:26.626998 72314 storage/queue.go:930  [replicate] purgatory is now empty
I190118 08:07:26.678598 72591 util/stop/stopper.go:536  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
W190118 08:07:26.695473 72797 storage/store.go:1465  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W190118 08:07:26.695815 72799 storage/store.go:1465  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W190118 08:07:26.696732 72806 internal/client/txn.go:520  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190118 08:07:26.697093 72806 storage/node_liveness.go:452  [hb] failed node liveness heartbeat: node unavailable; try another peer
I190118 08:07:26.698637 72591 util/stop/stopper.go:536  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
W190118 08:07:26.699269 72798 storage/store.go:1465  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
--- FAIL: TestLeaseNotUsedAfterRestart (0.28s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,30 pro=0.000000123,31
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,30 pro=0.000000123,31

```

SHA: https://github.com/cockroachdb/cockroach/commits/e09c33ffc49f15b4909d4f4509b96d1c3c36a956

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1125070&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190205 08:38:12.774111 72740 storage/store.go:1478  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190205 08:38:12.774790 72634 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:42211" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190205 08:38:12.819978 72634 storage/client_replica_test.go:1126  restarting
I190205 08:38:12.825733 72879 util/stop/stopper.go:536  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
3      [async] storage.pendingLeaseRequest: requesting lease
I190205 08:38:12.825895 72879 util/stop/stopper.go:536  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
2      [async] storage.pendingLeaseRequest: requesting lease
W190205 08:38:12.825919 72876 internal/client/txn.go:520  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190205 08:38:12.825960 72876 storage/node_liveness.go:452  [hb] failed node liveness heartbeat: node unavailable; try another peer
I190205 08:38:12.826007 72879 util/stop/stopper.go:536  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
I190205 08:38:12.826044 72879 util/stop/stopper.go:536  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
W190205 08:38:12.826138 72867 storage/store.go:1478  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: [NotLeaseHolderError] r1: replica (n1,s1):1 not lease holder; lease holder unknown
--- FAIL: TestLeaseNotUsedAfterRestart (0.07s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,27 pro=0.000000123,28
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,27 pro=0.000000123,28

```

SHA: https://github.com/cockroachdb/cockroach/commits/40e403544d60b1a44b8b1ed961a817c77d67aa31

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1131582&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190209 06:32:04.193073 74192 storage/store.go:1482  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190209 06:32:04.202177 74117 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:34809" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190209 06:32:04.252422 74117 storage/client_replica_test.go:1126  restarting
I190209 06:32:04.252472 74117 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] intent_resolver_batcher
I190209 06:32:04.257305 74342 util/stop/stopper.go:536  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
3      [async] storage.pendingLeaseRequest: requesting lease
1      [async] storage.replicate: processing replica
1      [async] intent_resolver_batcher
I190209 06:32:04.257345 74342 util/stop/stopper.go:536  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
3      [async] storage.pendingLeaseRequest: requesting lease
1      [async] storage.replicate: processing replica
I190209 06:32:04.257378 74342 util/stop/stopper.go:536  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
2      [async] storage.pendingLeaseRequest: requesting lease
1      [async] storage.replicate: processing replica
I190209 06:32:04.257428 74342 util/stop/stopper.go:536  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
2      [async] storage.pendingLeaseRequest: requesting lease
1      [async] storage.replicate: processing replica
I190209 06:32:04.257458 74342 util/stop/stopper.go:536  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
I190209 06:32:04.257485 74320 storage/queue.go:912  [replicate] purgatory is now empty
I190209 06:32:04.257497 74342 util/stop/stopper.go:536  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
I190209 06:32:04.257626 74342 util/stop/stopper.go:536  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
W190209 06:32:04.257656 74315 internal/client/txn.go:520  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190209 06:32:04.257677 74315 storage/node_liveness.go:452  [hb] failed node liveness heartbeat: node unavailable; try another peer
--- FAIL: TestLeaseNotUsedAfterRestart (0.09s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,26 pro=0.000000123,27
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,26 pro=0.000000123,27

```

SHA: https://github.com/cockroachdb/cockroach/commits/acba091f04f3d8ecabf51009bf394951fbd3643c

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1138046&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190214 06:53:17.763137 74200 storage/store.go:1478  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190214 06:53:17.764174 73805 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:44781" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190214 06:53:17.812378 73805 storage/client_replica_test.go:1126  restarting
I190214 06:53:17.814689 73805 util/stop/stopper.go:547  quiescing; tasks left:
1      [async] intent_resolver_batcher
I190214 06:53:17.819523 74120 util/stop/stopper.go:547  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
3      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_batcher
I190214 06:53:17.819550 74120 util/stop/stopper.go:547  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
3      [async] storage.pendingLeaseRequest: requesting lease
I190214 06:53:17.819613 74120 util/stop/stopper.go:547  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
2      [async] storage.pendingLeaseRequest: requesting lease
I190214 06:53:17.819653 74120 util/stop/stopper.go:547  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
I190214 06:53:17.819744 74358 storage/client_test.go:1318  [hb,txn=26291814,range-lookup=/Meta2/System/NodeLiveness/1/NULL] test clock advanced to: 2.800000125,0
I190214 06:53:17.819756 74120 util/stop/stopper.go:547  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
W190214 06:53:17.819799 74332 storage/store.go:1478  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: [NotLeaseHolderError] r2: replica (n1,s1):1 not lease holder; lease holder unknown
W190214 06:53:17.819957 74331 storage/store.go:1478  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
W190214 06:53:17.820004 74339 internal/client/txn.go:520  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190214 06:53:17.820033 74339 storage/node_liveness.go:452  [hb] failed node liveness heartbeat: node unavailable; try another peer
I190214 06:53:17.820092 74120 util/stop/stopper.go:547  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
W190214 06:53:17.820210 74330 storage/store.go:1478  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: [NotLeaseHolderError] r1: replica (n1,s1):1 not lease holder; lease holder unknown
--- FAIL: TestLeaseNotUsedAfterRestart (0.10s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,29 pro=0.000000123,30
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,29 pro=0.000000123,30

```

SHA: https://github.com/cockroachdb/cockroach/commits/06ccbf4c5c7d32328a07e82dc4327992084d130a

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1157943&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190301 10:06:49.885238 73299 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190301 10:06:49.903949 73209 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:42823" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190301 10:06:49.963021 73209 storage/client_replica_test.go:1126  restarting
I190301 10:06:49.970874 73209 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] storage.pendingLeaseRequest: requesting lease
1      storage.replicate: purgatory processing replica
1      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190301 10:06:49.971225 73209 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] storage.pendingLeaseRequest: requesting lease
1      storage.replicate: purgatory processing replica
1      mtc send
I190301 10:06:49.971927 73209 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.replicate: purgatory processing replica
1      mtc send
1      [async] storage.pendingLeaseRequest: requesting lease
I190301 10:06:49.973475 73209 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.replicate: purgatory processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
W190301 10:06:49.975576 73336 internal/client/txn.go:509  [s1,r5/1:/{Systemtse-Table/System…}] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190301 10:06:49.975954 73336 storage/node_liveness.go:775  [s1,r5/1:/{Systemtse-Table/System…}] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
I190301 10:06:49.979977 73209 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.replicate: purgatory processing replica
I190301 10:06:49.980547 73202 storage/queue.go:915  [replicate] purgatory is now empty
I190301 10:06:50.039956 73225 util/stop/stopper.go:546  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
2      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190301 10:06:50.040457 73225 util/stop/stopper.go:546  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
2      [async] storage.pendingLeaseRequest: requesting lease
I190301 10:06:50.041531 73225 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
W190301 10:06:50.042866 73428 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190301 10:06:50.043357 73428 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
I190301 10:06:50.043697 73225 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
W190301 10:06:50.044465 73421 storage/store.go:1485  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: [NotLeaseHolderError] r2: replica (n1,s1):1 not lease holder; lease holder unknown
W190301 10:06:50.060571 73419 storage/store.go:1485  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestLeaseNotUsedAfterRestart (0.31s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,23 pro=0.000000123,24
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,23 pro=0.000000123,24

```

SHA: https://github.com/cockroachdb/cockroach/commits/de1793532332fb64fca27cafe92d2481d900a5a0

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1160868&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190304 07:29:41.516356 73436 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190304 07:29:41.521854 73312 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:36989" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190304 07:29:41.587305 73312 storage/client_replica_test.go:1126  restarting
I190304 07:29:41.602687 73312 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190304 07:29:41.603015 73312 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_gc_batcher
I190304 07:29:41.603335 73312 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
I190304 07:29:41.603724 73312 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
I190304 07:29:41.668163 73457 util/stop/stopper.go:546  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
W190304 07:29:41.669625 73542 storage/store.go:1485  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: [NotLeaseHolderError] r1: replica (n1,s1):1 not lease holder; lease holder unknown
I190304 07:29:41.671246 73457 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      mtc send
1      [async] storage.replicate: processing replica
I190304 07:29:41.671750 73347 storage/client_test.go:1326  [hb,txn=11942446,range-lookup=/Meta2/System/NodeLiveness/1/NULL] test clock advanced to: 2.800000125,0
I190304 07:29:41.671842 73457 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.replicate: processing replica
W190304 07:29:41.672232 73544 storage/store.go:1485  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: [NotLeaseHolderError] r2: replica (n1,s1):1 not lease holder; lease holder unknown
I190304 07:29:41.672430 73457 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
W190304 07:29:41.678449 73551 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: aborted in distSender: context canceled
I190304 07:29:41.679428 73458 internal/client/txn.go:618  async rollback failed: node unavailable; try another peer
--- FAIL: TestLeaseNotUsedAfterRestart (0.25s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,33 pro=0.000000123,34
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,33 pro=0.000000123,34

```

SHA: https://github.com/cockroachdb/cockroach/commits/38bb1e7905b89f911bd74be4f5830217ffb7b343

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1168931&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190309 07:31:50.183471 515963 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190309 07:31:50.185260 515819 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:44853" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190309 07:31:50.237896 515819 storage/client_replica_test.go:1126  restarting
I190309 07:31:50.257706 515819 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190309 07:31:50.274388 516273 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190309 07:31:50.274429 516273 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] closedts-rangefeed-subscriber
W190309 07:31:50.275779 516262 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W190309 07:31:50.275868 516263 storage/store.go:1485  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W190309 07:31:50.275933 516261 storage/store.go:1485  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestLeaseNotUsedAfterRestart (0.13s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,26 pro=0.000000123,27
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,26 pro=0.000000123,27

```

SHA: https://github.com/cockroachdb/cockroach/commits/5f7de3c35348e847dc0d4ab1ba8d76574e1706c2

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1170158&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190310 06:39:11.743156 545533 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190310 06:39:11.745315 545463 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:44511" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190310 06:39:11.807818 545463 storage/client_replica_test.go:1126  restarting
I190310 06:39:11.818870 545463 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190310 06:39:11.819014 545463 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] intent_resolver_ir_batcher
I190310 06:39:11.819112 545463 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] storage.replicate: processing replica
I190310 06:39:11.844372 545463 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
W190310 06:39:11.845071 545726 internal/client/txn.go:509  [s1,r5/1:/{Systemtse-Table/System…}] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190310 06:39:11.845127 545726 storage/node_liveness.go:775  [s1,r5/1:/{Systemtse-Table/System…}] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
I190310 06:39:11.845157 545463 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
W190310 06:39:11.846013 545726 internal/client/txn.go:509  [s1,r5/1:/{Systemtse-Table/System…}] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
E190310 06:39:11.846070 545726 storage/replica_range_lease.go:292  [s1,r5/1:/{Systemtse-Table/System…}] node unavailable; try another peer
I190310 06:39:11.846134 545463 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
I190310 06:39:11.957305 546016 util/stop/stopper.go:546  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
3      [async] storage.pendingLeaseRequest: requesting lease
1      [async] storage.replicate: processing replica
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190310 06:39:11.957363 546100 storage/queue.go:915  [replicate] purgatory is now empty
W190310 06:39:11.957539 546074 storage/store.go:1485  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: [NotLeaseHolderError] r1: replica (n1,s1):1 not lease holder; lease holder unknown
W190310 06:39:11.957662 546075 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190310 06:39:11.957705 546016 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
I190310 06:39:11.958040 546016 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
--- FAIL: TestLeaseNotUsedAfterRestart (0.30s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,26 pro=0.000000123,27
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,26 pro=0.000000123,27

```

SHA: https://github.com/cockroachdb/cockroach/commits/9d058d53c8a82fceb2205f1827c26f1bf36c32ba

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1172693&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190312 06:24:21.230968 559391 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190312 06:24:21.232703 559316 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:40037" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190312 06:24:21.277438 559316 storage/client_replica_test.go:1126  restarting
I190312 06:24:21.279568 559316 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190312 06:24:21.279634 559316 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
I190312 06:24:21.279664 559316 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
I190312 06:24:21.279800 559316 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
I190312 06:24:21.280135 559497 storage/queue.go:915  [replicate] purgatory is now empty
I190312 06:24:21.303138 559260 util/stop/stopper.go:546  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190312 06:24:21.303179 559260 util/stop/stopper.go:546  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] intent_resolver_gc_batcher
I190312 06:24:21.303455 559260 util/stop/stopper.go:546  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_gc_batcher
I190312 06:24:21.303481 559260 util/stop/stopper.go:546  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
W190312 06:24:21.303806 559601 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190312 06:24:21.303835 559601 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
I190312 06:24:21.303856 559260 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
I190312 06:24:21.304298 559260 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.replicate: processing replica
I190312 06:24:21.304331 559260 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
I190312 06:24:21.304452 559467 storage/queue.go:915  [replicate] purgatory is now empty
--- FAIL: TestLeaseNotUsedAfterRestart (0.10s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,23 pro=0.000000123,24
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,23 pro=0.000000123,24

```

SHA: https://github.com/cockroachdb/cockroach/commits/70e3468e7ed5fb30b10eaaf972acbb0f16099d01

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1179564&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190315 07:27:25.042150 618158 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190315 07:27:25.054134 617938 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:45941" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190315 07:27:25.092896 617938 storage/client_replica_test.go:1126  restarting
I190315 07:27:25.095592 617938 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190315 07:27:25.095621 617938 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190315 07:27:25.118181 618343 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190315 07:27:25.118309 618343 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] closedts-rangefeed-subscriber
W190315 07:27:25.118780 618331 storage/store.go:1485  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W190315 07:27:25.118862 618333 storage/store.go:1485  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W190315 07:27:25.118880 618332 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
I190315 07:27:25.119101 618343 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
--- FAIL: TestLeaseNotUsedAfterRestart (0.17s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,41 pro=0.000000123,42
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,41 pro=0.000000123,42

```

SHA: https://github.com/cockroachdb/cockroach/commits/c2939ec9a4f15b7fb8683a5805deeb241953e7aa

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1183174&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190318 06:03:56.191658 698158 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190318 06:03:56.192774 697874 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:42253" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190318 06:03:56.243843 697874 storage/client_replica_test.go:1126  restarting
I190318 06:03:56.244467 697874 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190318 06:03:56.244653 697874 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] intent_resolver_ir_batcher
I190318 06:03:56.244764 697874 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
I190318 06:03:56.244877 697874 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
W190318 06:03:56.245109 698277 internal/client/txn.go:509  [s1,r5/1:/{Systemtse-Table/System…}] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190318 06:03:56.245140 698277 storage/node_liveness.go:775  [s1,r5/1:/{Systemtse-Table/System…}] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
W190318 06:03:56.245255 698277 internal/client/txn.go:509  [s1,r5/1:/{Systemtse-Table/System…}] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
E190318 06:03:56.245292 698277 storage/replica_range_lease.go:292  [s1,r5/1:/{Systemtse-Table/System…}] node unavailable; try another peer
I190318 06:03:56.259760 698370 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190318 06:03:56.260582 698061 storage/queue.go:915  [replicate] purgatory is now empty
I190318 06:03:56.260675 698370 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      mtc send
1      [async] storage.pendingLeaseRequest: requesting lease
I190318 06:03:56.260917 698370 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
W190318 06:03:56.261220 698366 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
I190318 06:03:56.261447 698387 internal/client/txn.go:618  async rollback failed: node unavailable; try another peer
--- FAIL: TestLeaseNotUsedAfterRestart (0.09s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,26 pro=0.000000123,28
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,26 pro=0.000000123,28

```

SHA: https://github.com/cockroachdb/cockroach/commits/dfa23c01e4ea39b19ca8b2e5c8a4e7cf9b9445f4

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1190629&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190321 06:54:09.438123 669998 storage/store.go:1487  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190321 06:54:09.440638 669874 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:35639" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190321 06:54:09.498796 669874 storage/client_replica_test.go:1126  restarting
I190321 06:54:09.500139 669874 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190321 06:54:09.500185 669874 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] intent_resolver_ir_batcher
I190321 06:54:09.500238 669874 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] storage.replicate: processing replica
I190321 06:54:09.503860 669874 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
W190321 06:54:09.507023 669839 internal/client/txn.go:509  [s1,r5/1:/{Systemtse-Table/System…}] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190321 06:54:09.507072 669839 storage/node_liveness.go:775  [s1,r5/1:/{Systemtse-Table/System…}] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
I190321 06:54:09.507504 669874 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
I190321 06:54:09.509626 670147 storage/queue.go:915  [replicate] purgatory is now empty
I190321 06:54:09.518921 670263 util/stop/stopper.go:546  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190321 06:54:09.519999 670263 util/stop/stopper.go:546  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
1      mtc send
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190321 06:54:09.520273 670260 storage/client_test.go:1327  [hb,txn=c57ad8e6,range-lookup=/Meta2/System/NodeLiveness/1/NULL] test clock advanced to: 2.800000125,0
I190321 06:54:09.520351 670263 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190321 06:54:09.520467 670263 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
W190321 06:54:09.520878 670257 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190321 06:54:09.520969 670257 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
--- FAIL: TestLeaseNotUsedAfterRestart (0.16s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,13 pro=0.000000123,14
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,13 pro=0.000000123,14

```

SHA: https://github.com/cockroachdb/cockroach/commits/a200cea4368ec90aaee12337d7ab5f9ca555108f

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1192830&tab=buildLog

```
I190322 06:29:20.740562 781373 storage/client_replica_test.go:1126  restarting
I190322 06:29:20.740616 781373 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190322 06:29:20.740647 781373 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] intent_resolver_ir_batcher
I190322 06:29:20.740674 781373 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] storage.replicate: processing replica
I190322 06:29:20.740713 781373 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
W190322 06:29:20.741187 781621 internal/client/txn.go:509  [s1,r5/1:/{Systemtse-Table/System…}] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190322 06:29:20.741217 781621 storage/node_liveness.go:775  [s1,r5/1:/{Systemtse-Table/System…}] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
W190322 06:29:20.741342 781621 internal/client/txn.go:509  [s1,r5/1:/{Systemtse-Table/System…}] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
E190322 06:29:20.741365 781621 storage/replica_range_lease.go:292  [s1,r5/1:/{Systemtse-Table/System…}] node unavailable; try another peer
I190322 06:29:20.750984 781722 util/stop/stopper.go:546  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
3      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190322 06:29:20.751023 781722 util/stop/stopper.go:546  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
3      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] intent_resolver_ir_batcher
I190322 06:29:20.751222 781722 util/stop/stopper.go:546  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] intent_resolver_ir_batcher
W190322 06:29:20.751430 781712 storage/store.go:1487  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: [NotLeaseHolderError] r2: replica (n1,s1):1 not lease holder; lease holder unknown
I190322 06:29:20.751460 781722 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
I190322 06:29:20.751478 781722 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
W190322 06:29:20.751578 781719 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190322 06:29:20.751601 781719 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
I190322 06:29:20.758450 781722 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
--- FAIL: TestLeaseNotUsedAfterRestart (0.09s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,3 pro=0.000000123,4
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,3 pro=0.000000123,4

```

SHA: https://github.com/cockroachdb/cockroach/commits/9399d559ae196e5cf2ad122195048ff9115ab56a

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1194690&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190323 06:13:12.673216 489701 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190323 06:13:12.680047 489565 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:46297" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190323 06:13:12.747607 489565 storage/client_replica_test.go:1126  restarting
I190323 06:13:12.752001 489565 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 06:13:12.752533 489565 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 06:13:12.752605 489565 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
W190323 06:13:12.771543 489946 internal/client/txn.go:509  [s1,r5/1:/{Systemtse-Table/System…}] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190323 06:13:12.771585 489946 storage/node_liveness.go:775  [s1,r5/1:/{Systemtse-Table/System…}] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
I190323 06:13:12.782586 489565 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
I190323 06:13:12.783218 489873 storage/queue.go:915  [replicate] purgatory is now empty
I190323 06:13:12.799942 490116 util/stop/stopper.go:546  quiescing; tasks left:
3      storage.Replica: acquiring lease to gossip
3      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
W190323 06:13:12.800677 490091 storage/store.go:1485  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: [NotLeaseHolderError] r2: replica (n1,s1):1 not lease holder; lease holder unknown
I190323 06:13:12.800776 490116 util/stop/stopper.go:546  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
2      [async] storage.pendingLeaseRequest: requesting lease
1      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
W190323 06:13:12.800862 490089 storage/store.go:1485  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: [NotLeaseHolderError] r1: replica (n1,s1):1 not lease holder; lease holder unknown
I190323 06:13:12.801418 490116 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] intent_resolver_gc_batcher
I190323 06:13:12.801454 490116 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
W190323 06:13:12.802382 490098 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
I190323 06:13:12.802557 490117 internal/client/txn.go:618  async rollback failed: node unavailable; try another peer
--- FAIL: TestLeaseNotUsedAfterRestart (0.23s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,23 pro=0.000000123,24
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,23 pro=0.000000123,24

```

SHA: https://github.com/cockroachdb/cockroach/commits/ec89b45cea7e8a6dd92a9bfd60e0cc06842e06d8

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1197532&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190325 05:55:17.482170 661401 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190325 05:55:17.489410 661312 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:38289" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190325 05:55:17.533142 661312 storage/client_replica_test.go:1126  restarting
I190325 05:55:17.534766 661312 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190325 05:55:17.534827 661312 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
I190325 05:55:17.534916 661312 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
I190325 05:55:17.534975 661312 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
I190325 05:55:17.535338 661312 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
I190325 05:55:17.535465 661437 storage/queue.go:915  [replicate] purgatory is now empty
I190325 05:55:17.544365 661521 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190325 05:55:17.544409 661521 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] closedts-rangefeed-subscriber
I190325 05:55:17.544423 661521 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
W190325 05:55:17.546236 661637 storage/store.go:1485  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W190325 05:55:17.546706 661646 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
I190325 05:55:17.548789 661313 internal/client/txn.go:618  async rollback failed: node unavailable; try another peer
--- FAIL: TestLeaseNotUsedAfterRestart (0.10s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,27 pro=0.000000123,28
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,27 pro=0.000000123,28

```

SHA: https://github.com/cockroachdb/cockroach/commits/7f8a0969e8e9eb7e9fc0d2fe96e03849d30dd561

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1200077&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190326 05:39:32.151320 737991 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190326 05:39:32.153123 737851 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:41681" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190326 05:39:32.221454 737851 storage/client_replica_test.go:1126  restarting
I190326 05:39:32.225469 737851 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 05:39:32.225501 737851 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190326 05:39:32.245301 737872 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 05:39:32.245367 737872 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
W190326 05:39:32.245940 738367 storage/store.go:1485  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W190326 05:39:32.247550 738366 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
--- FAIL: TestLeaseNotUsedAfterRestart (0.13s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,41 pro=0.000000123,42
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,41 pro=0.000000123,42

```

SHA: https://github.com/cockroachdb/cockroach/commits/3aadd20bbf0940ef65f8b2cdcda498401ba5d9c6

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1202388&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190327 05:46:59.416264 747100 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190327 05:46:59.434713 746990 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:37361" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190327 05:46:59.477539 746990 storage/client_replica_test.go:1126  restarting
I190327 05:46:59.510611 746990 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190327 05:46:59.510642 746990 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190327 05:46:59.528997 747551 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
W190327 05:46:59.542917 747540 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W190327 05:46:59.543729 747539 storage/store.go:1485  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W190327 05:46:59.543776 747541 storage/store.go:1485  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
--- FAIL: TestLeaseNotUsedAfterRestart (0.18s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,23 pro=0.000000123,24
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,23 pro=0.000000123,24

```

SHA: https://github.com/cockroachdb/cockroach/commits/a1e6e9decc9dec15a32bbb6d30efc67ca45a532a

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1205321&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190328 06:10:53.756287 801787 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190328 06:10:53.758638 801583 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:34821" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190328 06:10:53.809350 801583 storage/client_replica_test.go:1126  restarting
I190328 06:10:53.811825 801583 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190328 06:10:53.811864 801583 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190328 06:10:53.816551 802014 util/stop/stopper.go:546  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
2      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190328 06:10:53.816587 802014 util/stop/stopper.go:546  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
2      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] closedts-rangefeed-subscriber
I190328 06:10:53.816959 802014 util/stop/stopper.go:546  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] closedts-rangefeed-subscriber
I190328 06:10:53.816992 802014 util/stop/stopper.go:546  quiescing; tasks left:
2      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] closedts-rangefeed-subscriber
I190328 06:10:53.817030 802014 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] closedts-rangefeed-subscriber
I190328 06:10:53.817050 802014 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
W190328 06:10:53.817963 802082 storage/store.go:1485  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
I190328 06:10:53.819167 802014 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
W190328 06:10:53.819997 802081 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
--- FAIL: TestLeaseNotUsedAfterRestart (0.12s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,33 pro=0.000000123,34
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,33 pro=0.000000123,34

```

SHA: https://github.com/cockroachdb/cockroach/commits/17565100d1e7c66341e6db3e39bb66202958cb81

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1205349&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190328 06:20:53.058273 807724 storage/store.go:1497  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190328 06:20:53.063830 807568 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:39935" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190328 06:20:53.105167 807568 storage/client_replica_test.go:1126  restarting
I190328 06:20:53.107728 807568 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190328 06:20:53.116445 807839 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      mtc send
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190328 06:20:53.116486 807839 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      mtc send
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
W190328 06:20:53.116529 807940 storage/store.go:1497  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W190328 06:20:53.116562 807941 storage/store.go:1497  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
I190328 06:20:53.116656 807839 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      mtc send
1      [async] intent_resolver_ir_batcher
W190328 06:20:53.116966 807948 internal/client/txn.go:510  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190328 06:20:53.116998 807948 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
I190328 06:20:53.117015 807839 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
--- FAIL: TestLeaseNotUsedAfterRestart (0.10s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,7 pro=0.000000123,8
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,7 pro=0.000000123,8

```

SHA: https://github.com/cockroachdb/cockroach/commits/6e0d7081547ee76e52c5f798b78ff739feec2d06

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1208195&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190329 06:22:02.340981 493051 storage/store.go:1485  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190329 06:22:02.353529 492965 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:32877" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190329 06:22:02.390704 492965 storage/client_replica_test.go:1126  restarting
I190329 06:22:02.400484 492965 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190329 06:22:02.400560 492965 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_gc_batcher
I190329 06:22:02.400703 492965 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
I190329 06:22:02.414853 492965 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
W190329 06:22:02.415415 493170 internal/client/txn.go:509  [s1,r5/1:/{Systemtse-Table/System…}] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190329 06:22:02.415453 493170 storage/node_liveness.go:775  [s1,r5/1:/{Systemtse-Table/System…}] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
I190329 06:22:02.415867 492965 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.replicate: processing replica
I190329 06:22:02.416413 493220 storage/queue.go:915  [replicate] purgatory is now empty
I190329 06:22:02.427798 493324 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190329 06:22:02.427834 493324 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
I190329 06:22:02.427862 493324 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
W190329 06:22:02.428208 493317 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190329 06:22:02.428260 493317 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
W190329 06:22:02.428302 493308 storage/store.go:1485  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I190329 06:22:02.428359 493324 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
--- FAIL: TestLeaseNotUsedAfterRestart (0.16s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,27 pro=0.000000123,28
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,27 pro=0.000000123,28

```

SHA: https://github.com/cockroachdb/cockroach/commits/23f9707873abbd2de91a42055535529d7ff296ce

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1210240&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190330 05:42:22.963977 456377 storage/store.go:1489  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190330 05:42:23.024334 456230 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:35321" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
W190330 05:42:23.025684 456377 storage/store.go:1489  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190330 05:42:23.134322 456230 storage/client_replica_test.go:1126  restarting
I190330 05:42:23.148987 456230 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190330 05:42:23.150223 456230 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190330 05:42:23.174072 457095 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] storage.pendingLeaseRequest: requesting lease
1      storage.Replica: acquiring lease to gossip
1      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
W190330 05:42:23.174145 457084 storage/store.go:1489  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
I190330 05:42:23.174934 456972 storage/client_test.go:1325  [hb,txn=4ec7dcc2,range-lookup=/Meta2/System/NodeLiveness/1/NULL] test clock advanced to: 2.800000125,0
W190330 05:42:23.175263 457092 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190330 05:42:23.175304 457092 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
I190330 05:42:23.176286 457095 util/stop/stopper.go:546  quiescing; tasks left:
1      storage.Replica: acquiring lease to gossip
1      [async] storage.pendingLeaseRequest: requesting lease
W190330 05:42:23.176448 457083 storage/store.go:1489  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W190330 05:42:23.176876 457085 storage/store.go:1489  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: [NotLeaseHolderError] r2: replica (n1,s1):1 not lease holder; lease holder unknown
--- FAIL: TestLeaseNotUsedAfterRestart (0.38s)
    client_replica_test.go:1151: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,23 pro=0.000000123,24
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,23 pro=0.000000123,24

```

SHA: https://github.com/cockroachdb/cockroach/commits/877ebd1ece299b9ee621aa0d091657621593d844

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1216063&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190402 06:34:33.741801 542898 storage/store.go:1489  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190402 06:34:33.747120 542759 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:35877" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190402 06:34:33.830013 542759 storage/client_replica_test.go:1129  restarting
I190402 06:34:33.880539 542759 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190402 06:34:33.965908 543639 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
W190402 06:34:33.967188 543628 storage/store.go:1489  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W190402 06:34:33.967228 543629 storage/store.go:1489  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W190402 06:34:33.967253 543627 storage/store.go:1489  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestLeaseNotUsedAfterRestart (0.34s)
    client_replica_test.go:1154: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,31 pro=0.000000123,32
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,31 pro=0.000000123,32

```

SHA: https://github.com/cockroachdb/cockroach/commits/759f4076aced721bdfbb16bafb55f4662182e519

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestLeaseNotUsedAfterRestart PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1220605&tab=buildLog

```
=== RUN   TestLeaseNotUsedAfterRestart
W190404 06:16:55.044001 490623 storage/store.go:1501  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I190404 06:16:55.045736 490526 gossip/gossip.go:392  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:37199" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I190404 06:16:55.106066 490526 storage/client_replica_test.go:1129  restarting
I190404 06:16:55.107363 490526 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
W190404 06:16:55.107642 490808 internal/client/txn.go:509  [s1,r5/1:/{Systemtse-Table/System…}] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190404 06:16:55.107688 490808 storage/node_liveness.go:775  [s1,r5/1:/{Systemtse-Table/System…}] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
I190404 06:16:55.108040 490778 storage/queue.go:915  [replicate] purgatory is now empty
I190404 06:16:55.120216 490893 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190404 06:16:55.120252 490893 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] closedts-rangefeed-subscriber
I190404 06:16:55.121287 490893 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
--- FAIL: TestLeaseNotUsedAfterRestart (0.12s)
    client_replica_test.go:1154: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,31 pro=0.000000123,33
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,31 pro=0.000000123,33

```

Got a really fast repro cycle here:

```
--- FAIL: TestLeaseNotUsedAfterRestart (0.12s)
    client_replica_test.go:1155: lease was not replaced:
        prev: repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,10 pro=0.000000123,11
        now:  repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=0.900000123,10 pro=0.000000123,11
FAIL


ERROR: exit status 1

context canceled
Error:  exit status 1

0 runs completed, 0 failures, over 2s
```

Gonna see what this is all about.

--- a/pkg/storage/client_replica_test.go
+++ b/pkg/storage/client_replica_test.go
@@ -1077,8 +1077,6 @@ func TestLeaseMetricsOnSplitAndTransfer(t *testing.T) {
 func TestLeaseNotUsedAfterRestart(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
-	t.Skip("https://github.com/cockroachdb/cockroach/issues/34111")
-
 	ctx := context.Background()
 
 	sc := storage.TestStoreConfig(nil)
@@ -1101,17 +1099,14 @@ func TestLeaseNotUsedAfterRestart(t *testing.T) {
 	defer mtc.Stop()
 	mtc.Start(t, 1)
 
+	key := []byte("a")
 	// Send a read, to acquire a lease.
-	getArgs := getArgs([]byte("a"))
+	getArgs := getArgs(key)
 	if _, err := client.SendWrapped(ctx, mtc.stores[0].TestSender(), getArgs); err != nil {
 		t.Fatal(err)
 	}
 
-	preRepl1, err := mtc.stores[0].GetReplica(1)
-	if err != nil {
-		t.Fatal(err)
-	}
-	preRestartLease, _ := preRepl1.GetLease()
+	preRestartLease, _ := mtc.stores[0].LookupReplica(key).GetLease()
 
 	mtc.manualClock.Increment(1E9)
 
@@ -1142,11 +1137,7 @@ func TestLeaseNotUsedAfterRestart(t *testing.T) {
 		t.Fatalf("read did not acquire a new lease")
 	}
 
-	postRepl1, err := mtc.stores[0].GetReplica(1)
-	if err != nil {
-		t.Fatal(err)
-	}
-	postRestartLease, _ := postRepl1.GetLease()
+	postRestartLease, _ := mtc.stores[0].LookupReplica(key).GetLease()
 
 	// Verify that not only is a new lease requested, it also gets a new sequence
 	// number. This makes sure that previously proposed commands actually fail at

